numpy/lib/function_base.py

```diff
@@ -1320,6 +1320,477 @@ def gradient(f, *varargs, **kwargs):
     else:
         return outvals
 
+def divergence(v,*varargs,**kwargs):
+    """
+    Return the divergence of an N-dimensional vector field of N 
+    components, each of dimension N.
+
+    The divergence is computed using second order accurate central
+    differences in the interior and either first differences or second
+    order accurate one-sides (forward or backwards) differences at the
+    boundaries. The returned gradient hence has the same shape as the
+    input array.
+    
+    Parameters
+    ----------  
+    v : list of numpy arrays
```

Comment:
Wouldn't it make more sense to make this an array-like? Here, one can choose which dimension to be that of the axes; I think the first one is fine.  In that case, one could just do
```
v = np.asanyarray(v)
```
and it would even cover the case of having a list of numpy arrays. Another advantage would be that the tests for the sizes to be the same could be omitted, since that would be done implicitly already.